### PR TITLE
Upgrades to 5.6.16

### PR DESCRIPTION
**Pull request by devlinmr**
Upgrades binaries to 5.6.16
Removes searchguard ssl plugin
Removes elasticsearch-cloud-kubernetes plugin
Switches DISCOVERY_TYPE to file and adds unicast_hosts.txt instead.
Removes deprecated NODE_LOCAL setting
Adds an env var to control `bootstrap.memory_lock`: MEM_LOCK
Updates deprecated s3 cloud settings to `s3.client.default` equivalents.

### CODE REVIEW

File: config/elasticsearch.yml [hunk 1]
```diff
@@ -21,9 +20,9 @@ http.enabled: ${HTTP_ENABLE}
 http.compression: true
 http.bind_host: ${HTTP_BIND_HOST}
 
-cloud.kubernetes.service: ${KUBERNETES_SERVICE}
-cloud.kubernetes.namespace: ${KUBERNETES_NAMESPACE}
-discovery.zen.hosts_provider: ${DISCOVERY_TYPE}
+s3.client.default.access_key: ${CLOUD_AWS_S3_ACCESS_KEY}
+s3.client.default.secret_key: ${CLOUD_AWS_S3_SECRET_KEY}
```
Comment (marcinc) [inline config/elasticsearch.yml]: Maybe move those to where the previous s3 config options were. Is this format now enforced / supported in 5.6?
What about region?

File: run.sh [hunk 2]
```diff
@@ -1,10 +1,14 @@
 #!/bin/bash
 
+export UNICAST_HOSTS_FILE=${UNICAST_HOSTS_FILE:=/hosts/unicast_hosts.txt}
+
+[ -f $UNICAST_HOSTS_FILE ] && cp $UNICAST_HOSTS_FILE /elasticsearch/config/unicast_hosts.txt
```
Comment (marcinc) [inline run.sh]: Should we use headless service for nodes discovery as suggested in https://github.com/fabric8io/elasticsearch-cloud-kubernetes#kubernetes-cloud-plugin-for-elasticsearch?

File: config/elasticsearch.yml (hunk 1 above)
Comment (devlinmr) [reply config/elasticsearch.yml]: Done.

**Review by marcinc — APPROVED** [review]
Please quash commits. LG.

**Review by marcinc — APPROVED** [review]
LGTM. Assuming it works OK.

**Pull request merged**